### PR TITLE
B fix linking

### LINKED ISSUE #6: problem with linked lib when not using sudo (using ros_grant)
output is:

```
/opt/ros/hydro/lib/ros_ethercat_loop/ros_ethercat_loop: error while loading shared libraries: libcontroller_manager.so: cannot open shared object file: No such file or directory
```

### PR DESCRIPTION
**Pull request by toliver**
Fixes #6 

This is pending testing with installed debs, as it is then that the problem shows. I'm not sure how to generate the same conditions without actually doing a release.

### CODE REVIEW

File: ros_ethercat_model/cmake/ros_ethercat_model-extras.cmake.em [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+function(ros_enable_rpath target)
```
Comment (toliver) [inline ros_ethercat_model/cmake/ros_ethercat_model-extras.cmake.em:2]: This has been taken as-is from:
https://github.com/PR2/pr2_mechanism/blob/hydro-devel/pr2_hardware_interface/cmake/pr2_hardware_interface-extras.cmake.em

**Comment by shadowmanos** [comment]
Yes it can only be tested after release :-( I believe the problem is created at the ROS build farm.

**Comment by shadowmanos** [comment]
Why are ros_ethercat_model and ros_ethercat_hardware involved? The linking problem is at ros_ethercat_loop package and the ros_ethercat_loop executable

**Comment by toliver** [comment]
No, actually any library that has to find other libraries need to have the rpath set (provided that is used from an executable using capabilities).
- ros_ethercat_loop needs to find controller_manager and ros_ethercat_harware
- ros_ethercat_hardware needs to find ros_ethercat_eml

I decided to place the new cmake function in ros_ethercat_model, so that both the other packages can use it. Maybe it would be better to place it somewhere else. A more general place, as this function might be needed in other packages, like controllers that are loaded in runtime and depend on some ROS shared libraries.

**Comment by markpitchless** [comment]
Once I have the time to check out this new buildbot-ros thing that @ugo saw at roscon we will have local debian package generation again. Our current setup only supports groovy I think!

**Comment by toliver** [comment]
Ping

**Comment by guihomework** [comment]
I had exactly this error after installing (catkin_make install) because rpath are removed in install folder by default by CMake. I currently compile the indigo-devel branch in hydro and this fix works on my side, with pr2-grant (did not use ros_grant yet)
Why is this not merged ? Definitely for indigo-devel branch

**Pull request merged**